plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1332,6 +1334,84 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
         }
     }
 
+    @Override
+    public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, String newColumnName)
+    {
+        DeltaLakeTableHandle table = (DeltaLakeTableHandle) tableHandle;
+        DeltaLakeColumnHandle deltaLakeColumn = (DeltaLakeColumnHandle) columnHandle;
+        String sourceColumnName = deltaLakeColumn.getBaseColumnName();
+
+        checkSupportedWriterVersion(session, table);
+        if (changeDataFeedEnabled(table.getMetadataEntry()) && CHANGE_DATA_FEED_COLUMN_NAMES.contains(newColumnName)) {
```

Comment:
I honestly don't know. As our delta lake connector doesnt support dropping columns I was not able to test dropping a column and then adding new one with the same name.  

On databricks renaming column breaks reading table changes. I got this error:
```

Error in SQL statement: DeltaUnsupportedOperationException: Retrieving table changes between version 0 and 5 failed because of an incompatible schema change.
Your read schema is {"type":"struct","fields":[{"name":"id","type":"integer","nullable":true,"metadata":{"delta.columnMapping.id":1,"delta.columnMapping.physicalName":"id"}},{"name":"new_name","type":"string","nullable":true,"metadata":{"delta.columnMapping.id":2,"delta.columnMapping.physicalName":"name"}}]} at version 5, but we found an incompatible schema change at version 3.
If possible, please query table changes separately from version 0 to 3 - 1, and from version 3 to 5.
```